lms/services/user.py

```diff
@@ -94,6 +94,54 @@ def _user_search_query(self, application_instance_id, user_id) -> Select:
 
         return query
 
+    def get_users(
+        self,
+        role_scope: RoleScope,
+        role_type: RoleType,
+        instructor_h_userid: str | None = None,
+    ) -> Select[tuple[User]]:
+        """
+        Get a query to fetch users.
+
+        :param role_scope: return only users with this LTI role scope.
+        :param role_type: return only users with this LTI role type.
+        :param instructor_h_userid: return only users that belongs to courses/assignments where the user instructor_h_userid is an instructor.
+        """
+        query = (
+            select(User.id)
+            .join(AssignmentMembership)
+            .join(LTIRole)
+            .where(LTIRole.scope == role_scope, LTIRole.type == role_type)
+        )
+
+        if instructor_h_userid:
+            query = query.where(
+                AssignmentMembership.assignment_id.in_(
+                    select(AssignmentMembership.assignment_id)
+                    .join(User)
+                    .join(LTIRole)
+                    .where(
+                        User.h_userid == instructor_h_userid,
+                        LTIRole.scope == RoleScope.COURSE,
+                        LTIRole.type == RoleType.INSTRUCTOR,
+                    )
+                )
+            )
+
+        # Deduplicate based on the row's h_userid taking the last updated one
+        query = query.distinct(User.h_userid).order_by(
```

Comment:
Similar pattern we had for courses, we deduplicate based on a column/order but return rows based on a different order.

We use a subquery to achieve that.